exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java

```diff
@@ -39,6 +42,22 @@
 
   private Adapter() {}
 
+  /**
+   * Groups {@link SpanData}'s by {@link Resource}.
+   *
+   * @param spans the list of spans to be grouped
+   * @return the map of grouped spans
+   */
+  static Map<Resource, List<SpanData>> groupByResource(Collection<SpanData> spans) {
+    Map<Resource, List<SpanData>> result = new HashMap<>();
+    for (SpanData spanData : spans) {
+      Resource resource = spanData.getResource();
+      List<SpanData> spanDataList = result.computeIfAbsent(resource, k -> new ArrayList<>());
+      spanDataList.add(spanData);
+    }
+    return result;
```

Comment:
Did you consider using a java 7 Stream with a `groupingBy` collector to do this work? It might read a little more clearly. It's a one-liner in this case:

`    return spans.stream().collect(Collectors.groupingBy(SpanData::getResource));`

Reply:
(and, given that it's a one-liner...you can probably just one-liner it in the calling code, rather than adding a function here.)

Reply:
done